sqlglot/parser.py

```diff
@@ -1287,23 +1288,30 @@ def _parse_lateral(self):
             return None
 
         if not this:
-            this = self._parse_function()
-
-        table_alias = self._parse_id_var(any_token=False)
+            this = self._parse_function() or self._parse_id_var(False)
```

Comment:
why are you hardcoding False here?

Reply:
Made it more consistent within its function